src/FlatSharp.Compiler/SchemaModel/PropertyFieldModel.cs

```diff
@@ -317,6 +317,7 @@ private string GetVectorTypeName(string innerType, string? keyType)
             VectorType.Memory => $"Memory<{innerType}>",
             VectorType.ReadOnlyMemory => $"ReadOnlyMemory<{innerType}>",
             VectorType.IIndexedVector => $"IIndexedVector<{keyType ?? "string"}, {innerType}>",
+            VectorType.UnityNativeArray => $"Unity.Collections.NativeArray<{innerType}>",

```

Comment:
One thing I think will be a problem is that there is no check here to enforce that the type of inner value is actually a value type (ref struct). My suggestion is twofold:

- Relax the generic type restrictions in the polyfill to allow any type (so you don't get compile errors using a reference type)
- Add validation to the new type model to ensure that the inner member is a...
   - Value type / primitive
   - Value struct
   
But I think we want to exclude tables here as well as strings and unions.